packages/xo-web/src/common/modal.js

```diff
@@ -254,6 +259,56 @@ export const confirm = ({ body, icon = 'alarm', title, strongConfirm }) =>
 
 // -----------------------------------------------------------------------------
 
+export const form = ({ body, defaultValue, title, size }) =>
+  new Promise((resolve, reject) => {
+    const formId = generateRandomId()
+    const buttons = [
+      {
+        btnStyle: 'primary',
+        label: _('formSave'),
+        form: formId,
+      },
+    ]
+
+    const Modal = class extends BaseComponent {
+      state = {
+        value: defaultValue,
+      }
+
+      get value () {
+        return this.state.value
+      }
+
+      render () {
+        return (
+          <form id={formId}>
```

Comment:
You could create the Modal class outside and pass the `formId` as a prop to avoid creating a new class everytime.